### PR TITLE
Fix runner not dealing with str as stdin

### LINKED ISSUE #7145: Q: how to pass a string as stdin to Runners' run()?
The docstring of the Runner's [run](https://github.com/datalad/datalad/blob/66a8b77c09af3b0c8871ef541a061596e8257e44/datalad/runner/runner.py#L69) method state the following about the ``stdin=`` parameter:
https://github.com/datalad/datalad/blob/66a8b77c09af3b0c8871ef541a061596e8257e44/datalad/runner/runner.py#L91-L98

However, if I provide a string to it (I tried toying around in the context of https://github.com/datalad/datalad/issues/6514), I get
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/adina/repos/datalad/datalad/runner/runnerthreads.py", line 172, in run
    succeeded = self.write(data)
  File "/home/adina/repos/datalad/datalad/runner/runnerthreads.py", line 246, in write
    written += os.write(
TypeError: a bytes-like object is required, not 'str'
```

I feel like I might be missing something fairly obvious. How can I provide a string as stdin to a ``runner.run()`` call?

### PR DESCRIPTION
**Pull request by bpoldrack**
Fixes #7145

### CODE REVIEW

**Comment by bpoldrack** [comment]
Does this look aright to you, @christian-monch ?

File: datalad/runner/nonasyncrunner.py [hunk 1]
```diff
@@ -333,7 +333,8 @@ def _locked_run(self) -> dict | _ResultGenerator:
             # enqueue the input that is already provided.
             self.write_stdin = True
             self.stdin_queue = Queue()
-            self.stdin_queue.put(self.stdin)
+            self.stdin_queue.put(self.stdin if isinstance(self.stdin, bytes)
+                                 else self.stdin.encode(encoding='utf-8'))
```
Comment (christian-monch) [inline datalad/runner/nonasyncrunner.py:337]: Basically correct, but I would use the protocol_kwargs encoding value, if it is set. Besides that, the indentation in this file has a different style and strings here are always delimited with double-quotes.

```suggestion
            self.stdin_queue.put(
                self.stdin
                if isinstance(self.stdin, bytes)
                else self.stdin.encode(
                    encoding=self.protocol_kwargs.get("encoding", "utf-8")))
```

**Review by christian-monch — APPROVED** [review]
Thx @bpoldrack 
LGTM with the suggested changes

**Comment by bpoldrack** [comment]
@christian-monch: Agree with your take in #7145. Force-pushed accordingly.

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.17.10`](https://github.com/datalad/datalad/releases/tag/0.17.10)

**Pull request merged**